ly/musicxml/ly2xml_mediator.py

```diff
@@ -353,6 +354,17 @@ def new_key(self, key_name, mode):
         else:
             self.current_attr.set_key(get_fifths(key_name, mode), mode)
 
+    def new_mark(self):
+        if self.bar is None:
+            self.new_bar()
+        if self.bar.has_music():
+            new_bar_attr = xml_objs.BarAttr()
+            new_bar_attr.set_mark(self.current_mark)
+            self.add_to_bar(new_bar_attr)
+        else:
+            self.current_attr.set_mark(self.current_mark)
+        self.current_mark = chr(ord(self.current_mark) + 1)
```

Comment:
Now it is better. It correctly skips `I` and changes to the correct mark sequence after an ending `Z`. Now it just needs to support the integer argument to set the mark manually.